xxx/regs.c

```diff
@@ -0,0 +1,46 @@
+/**
+ * get register info
+ */
+
+#include <stdint.h>
+#include "types.h"
+
+/* 32bit registers */
+
+i32 eax() {
+        /* by calling convention */
+}
+
+i32 ebx() {
+        asm("mov eax, ebx");
+}
+
+i32 ecx() {
+        asm("mov eax, ecx");
+}
+
+i32 edx() {
+        asm("mov eax, edx");
+}
+
+i32 eip() {
+        asm("__geteip: mov eax, [esp]\n"
+            "          ret\n"
+            "call __geteip");
+}
```

Comment:
That's the old school `asm` directive. C and asm was cool before gcc introduced the *extended inline asm* thing...